src/histolab/slide.py

```diff
@@ -489,23 +513,9 @@ def save_thumbnails(self, n: int = 0) -> None:
         """
         os.makedirs(self._processed_path, exist_ok=True)
         n = self.total_slides if (n > self.total_slides or n == 0) else n
-        for slide in self.slides[:n]:
+        for slide in list(self.__iter__())[:n]:
             slide.save_thumbnail()
 
-    @lazyproperty
-    def slides(self) -> List[Slide]:
```

Comment:
my suggestion was for `SlideSet` to be an iterable object, but I wouldn't remove the `slides` property

Reply:
why keep that property if slideset contains itself in its __iter__ magic method?

Reply:
I'm thinking about explicitly getting one slide:
```python
i_th = slideset.slides[i]
```
instead of
```python
i_th = list(slideset)[i]
```

but at this point, we could directly implement `__getitem__`, which I think is the best choice anyway

Reply:
yeah exactly i do like it 👍🏽

Reply:
good, go ahead ;)